aopy/preproc/neuropixel.py

```diff
@@ -48,4 +51,143 @@ def sync_neuropixel_ecube(raw_timestamp,on_times_np,off_times_np,on_times_ecube,
+        
+    Returns:
+        spike_times_unit (dict): spike times for each unit
+    '''
+    
+    spike_times_unit = {}
+    
+    for unit_label in np.unique(spike_label):
+        spike_times_unit[f'{unit_label}'] = spike_times[spike_label==unit_label.astype(int)]
+    
+    return spike_times_unit
+
+def parse_ksdata_entries(kilosort_dir, concat_data_dir):
+    '''
+    Parse concatenated data processed by kilosort into the task entries and save relevant data (spike_indices, spike_clusters, and ks_label)
+    
+    Args:
+        kilosort_dir (str): kilosort directory (ex. '/data/preprocessed/kilosort')
+        concat_data_dir (str): data directory that contains concatenated data and kilosort_output (ex. '2023-06-30_Neuropixel_ks_affi_bottom_port1')
+        
+    Returns
+        None
+    '''
+    
+    # Load kilosort data
+    kilosort_output = load_ks_output(kilosort_dir, concat_data_dir, flag='spike')
+    spike_indices = kilosort_output['spike_indices']
+    spike_clusters = kilosort_output['spike_clusters']
+    ks_label = kilosort_output['ks_label']
+    
+    # Load datasize of each entry and filename
+    data_path = os.path.join(kilosort_dir, concat_data_dir)
+    datasize_entry = np.load(os.path.join(data_path, 'datasize_entry.npy'))
+    task_paths = np.load(os.path.join(data_path, 'task_path.npy'))
+
+    nentry = datasize_entry.shape[0]
+
+    datasize_entry = np.cumsum(datasize_entry)
+    for idx in range(nentry):
+        # Get spike indices included in a corresponding entry
+        if idx == 0:
+            ientry_idx = spike_indices < datasize_entry[idx]
+        else:
+            ientry_idx = (spike_indices >= datasize_entry[idx-1])&(spike_indices < datasize_entry[idx])
+
+        # Subtract previous entries' datasize so that the first index of each entry can become 0.
+        spike_indices_entry = spike_indices[ientry_idx]
+        if idx > 0:
+            spike_indices_entry -= datasize_entry[idx-1]
+        
+        # Make new directory to save parsed data
+        taskid = task_paths[idx].split('_')[-1][2:]
+        task_save_dir = f'{concat_data_dir}_{taskid}'
+        task_save_path = os.path.join(kilosort_dir, task_save_dir)
+        if not os.path.exists(task_save_path):
+            os.makedirs(task_save_path)
+        
+        # save data
+        np.save(os.path.join(task_save_path, 'spike_indices_entry'), spike_indices_entry)
+        np.save(os.path.join(task_save_path, 'spike_clusters_entry'), spike_clusters[ientry_idx])
+        np.save(os.path.join(task_save_path, 'ks_label'), ks_label.astype('str'))
+        
+def concat_neuropixel_within_day(np_datadir, kilosort_dir, subject, date, ch_config_dir='/data/channel_config_np', port_number = 1):
+    '''
+    concatenate continuous.dat files of different sessions within day and subject
+    then save the concatenated continuous data into kilosort directory with datasize and id of each task entry

```

Comment:
Its probably worth mentioning that this function only concatenates data files with the same electrode configuration.